.github/workflows/release.yml

```diff
@@ -0,0 +1,165 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: The branch to release from
+        required: true
+        default: master
+      release-type:
+        description: The type of release to perform
+        required: true
+        default: GA
+        type: choice
+        options:
+          - GA
+          - RC
+          - Beta
+
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    name: Validate input parameters
+    outputs:
+      vtag: ${{ steps.vtag.outputs.vtag }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.branch }}
+      - name: Create version tag
+        id: vtag
+        run: |
+          PACKAGE_VERSION=$(sed -n 's/^ *"version": *"//p' package.json | tr -d '"' | tr -d ',' | tr -d '[[:space:]]')
+          VTAG="${PACKAGE_VERSION}.${{ github.event.inputs.release-type }}"
+          echo "vtag=${VTAG}" >> $GITHUB_ENV
+          echo "::set-output name=vtag::${VTAG}"
+      - name: Validate version
+        uses: actions/github-script@v5
+        env:
+          vtag: ${{ env.vtag }}
+        with:
+          script: |
+            const releaseMeta = github.rest.repos.listReleases.endpoint.merge({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+            });
+
+            const releases = await github.paginate(releaseMeta);
+
+            for (const release of releases) {
+              if (release.name === process.env.vtag) {
+                throw new Error(`${process.env.vtag} already exists`);
+              }
+            }
+            console.log(`${process.env.vtag} does not exist so continuing`)
+
+  android:
+    runs-on: ubuntu-18.04
+    needs: [validate]
+    env:
+      CCACHE_DIR: ${{ github.workspace }}/.ccache
+      USE_CCACHE: 1
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.branch }}
+      - name: Android build
+        uses: ./.github/actions/build-android
+        with:
+          node-version: '12.x'
+          java-version: '11'
+
+  ios:
+    runs-on: macos-11
+    name: iOS
+    needs: [validate]
+    env:
+      CCACHE_DIR: ${{ github.workspace }}/.ccache
+      USE_CCACHE: 1
+      DEVELOPER_DIR: /Applications/Xcode_13.0.app/Contents/Developer
+    steps:
+      - uses: actions/checkout@v2
+        with:
+            ref: ${{ github.event.inputs.branch }}
+      - name: iOS build
+        uses: ./.github/actions/build-ios
+        with:
+          node-version: '12.x'
+
+  package:
+    runs-on: macos-11
+    name: Package
+    env:
+      SDK_BUILD_CACHE_DIR: ${{ github.workspace }}/.native-modules
+      DEVELOPER_DIR: /Applications/Xcode_13.0.app/Contents/Developer
```

Comment:
Shouldn't we be using Xcode 13.2.1?

13.2.1 is the last Xcode version that supports macOS Big Sur 11.

13.3 requires macOS Monterey 12 and while GitHub doesn't support macOS 12 yet, it will soon: https://github.com/actions/virtual-environments/issues/3649#issuecomment-1074058969.

Reply:
Similar to the Node 16 comment, I left everything as it to avoid introducing new things that could break while I was refactoring all this. I can update this to the latest available Xcode if desired